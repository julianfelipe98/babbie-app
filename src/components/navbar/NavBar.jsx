import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import MenuItem from "@material-ui/core/MenuItem";
import Menu from "@material-ui/core/Menu";
import SearchIcon from "@material-ui/icons/Search";
import MoreIcon from "@material-ui/icons/MoreVert";
import { Button } from "@material-ui/core";
import { BottonNavegaton } from "./BottonNavegaton";
import logo from '../../assets/logo.svg'
const useStyles = makeStyles((theme) => ({
    grow: {
        flexGrow: 2,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    title: {
        display: "none",
        [theme.breakpoints.up("sm")]: {
            display: "block",
        },
    },

    inputRoot: {
        color: "inherit",
    },
    inputInput: {
        padding: theme.spacing(1, 1, 1, 0),
        // vertical padding + font size from searchIcon
        paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
        transition: theme.transitions.create("width"),
        width: "100%",
        [theme.breakpoints.up("md")]: {
            width: "20ch",
        },
    },
    sectionDesktop: {
        display: "none",
        [theme.breakpoints.up("md")]: {
            display: "flex",
        },
    },
    sectionMobile: {
        display: "flex",
        [theme.breakpoints.up("md")]: {
            display: "none",
        },
    },
    img: {
        width: "50%",
        height: "auto",
    },
}));

export const NavBar = () => {
    const classes = useStyles();
    const [mobileMoreAnchorEl, setMobileMoreAnchorEl] = React.useState(null);

    const isMobileMenuOpen = Boolean(mobileMoreAnchorEl);

    const handleMobileMenuClose = () => {
        setMobileMoreAnchorEl(null);
    };


    const handleMobileMenuOpen = (event) => {
        setMobileMoreAnchorEl(event.currentTarget);
    };


    const mobileMenuId = "primary-search-account-menu-mobile";
    const renderMobileMenu = (
        <Menu
            anchorEl={mobileMoreAnchorEl}
            anchorOrigin={{ vertical: "top", horizontal: "right" }}
            id={mobileMenuId}
            keepMounted
            transformOrigin={{ vertical: "top", horizontal: "right" }}
            open={isMobileMenuOpen}
            onClose={handleMobileMenuClose}
        >
            <MenuItem>
                <Button variant="outlined" color="primary" edge="end">
                    <SearchIcon />
                    Buscar historial medico
                </Button>
            </MenuItem>
        </Menu>
    );

    return (
        <div className={classes.grow}>
            <AppBar position="fixed" color="inherit">
                <Toolbar>
                    <Typography className={classes.title} variant="h6" noWrap>
                        <img
                            id="logo-img"
                            src={logo}
                            className={classes.img}
                            alt="logo-babbie"
                        ></img>
                    </Typography>
                    <div className={classes.grow} />
                    <BottonNavegaton />
                    <div className={classes.grow} />
                    <div className={classes.sectionDesktop}>
                        <Button variant="outlined" color="primary" edge="end">
                            <SearchIcon />
                            Buscar historial medico
                        </Button>
                    </div>
                    <div className={classes.sectionMobile}>
                        <IconButton
                            aria-label="show more"
                            aria-controls={mobileMenuId}
                            aria-haspopup="true"
                            onClick={handleMobileMenuOpen}
                            color="inherit"
                        >
                            <MoreIcon />
                        </IconButton>
                    </div>
                </Toolbar>
            </AppBar>
            {renderMobileMenu}
        </div>
    );
};
