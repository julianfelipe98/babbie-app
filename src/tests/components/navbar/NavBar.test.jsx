import { shallow } from "enzyme";
import { NavBar } from "../../../components/navbar/NavBar";
describe("NavBar component tests", () => {
  test("should render <NavBar/> in the correct way", () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
