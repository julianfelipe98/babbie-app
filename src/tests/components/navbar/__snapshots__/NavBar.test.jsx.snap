// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component tests should render <NavBar/> in the correct way 1`] = `
<div
  className="makeStyles-grow-1"
>
  <WithStyles(ForwardRef(AppBar))
    color="inherit"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        noWrap={true}
        variant="h6"
      >
        <img
          alt="logo-babbie"
          className="makeStyles-img-8"
          id="logo-img"
          src="logo.svg"
        />
      </WithStyles(ForwardRef(Typography))>
      <div
        className="makeStyles-grow-1"
      />
      <BottonNavegaton />
      <div
        className="makeStyles-sectionDesktop-6"
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          edge="end"
          variant="outlined"
        >
          <Memo([object Object]) />
          Buscar historial medico
        </WithStyles(ForwardRef(Button))>
      </div>
      <div
        className="makeStyles-sectionMobile-7"
      >
        <WithStyles(ForwardRef(IconButton))
          aria-controls="primary-search-account-menu-mobile"
          aria-haspopup="true"
          aria-label="show more"
          color="inherit"
          onClick={[Function]}
        >
          <Memo([object Object]) />
        </WithStyles(ForwardRef(IconButton))>
      </div>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    id="primary-search-account-menu-mobile"
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
  >
    <WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        edge="end"
        variant="outlined"
      >
        <Memo([object Object]) />
        Buscar historial medico
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</div>
`;
